lib/widgets/icons.dart

```diff
@@ -57,14 +57,17 @@ abstract final class ZulipIcons {
   /// The Zulip custom icon "read_receipts".
   static const IconData read_receipts = IconData(0xf10b, fontFamily: "Zulip Icons");
 
+  /// The Zulip custom icon "star".
+  static const IconData star = IconData(0xf10c, fontFamily: "Zulip Icons");
```

Comment:
icons: Add star custom icon
    
    This is a filled version of the Feather star icon.
    Copied from zulip web `web/shared/icons/star-filled.svg`
    which sourced it from:
      https://github.com/zulip/zulip/issues/25903

We're implementing a specific design that's intended for mobile. That design includes an icon, which I pointed to here:
https://github.com/zulip/zulip-flutter/pull/449#discussion_r1445575735

So we should use the icon from the design we're intending to implement. A similar icon in web might happen to be the same, or it might not — sometimes there are differences because it's a different context.